src/plugins/dashboard/public/application/listing/dashboard_unsaved_listing.tsx

```diff
@@ -125,50 +142,52 @@ export const DashboardUnsavedListing = ({ redirectTo }: { redirectTo: DashboardR
     [overlays, dashboardPanelStorage]
   );
 
-  useMount(() => {
-    setDashboardIds(dashboardPanelStorage.getDashboardIdsWithUnsavedChanges());
+  useEffect(() => {
+    return () => setMounted(false);
```

Comment:
This is run on every state change. If you wanted something like componentWillUnmount then it would have to be a bit different: 
https://github.com/streamich/react-use/blob/master/src/useUnmount.ts

But as I understand, we need this for request cancelation, then the best would be to handle it as part of the effect with the request


```
useEffect(() => {
   let canceled = false;
   request().then(() => {
     if (canceled) return;
   })

   return () => {
       canceled = true;
  }
})

```

Reply:
Wow, huh. I meant to pass in an empty dependency array there, which would have made it run only on unmount. I need to be more careful with these things. Thanks for noticing this!

That said, the approach of canceling within the useEffect is cleaner anyway. I have implemented it.